admiral/pkg/clusters/types_test.go

```diff
@@ -257,5 +262,184 @@ func TestHandleEventForGlobalTrafficPolicy(t *testing.T) {
+		LabelSet:                   &common.LabelSet{},
+		EnableSAN:                  true,
+		SANPrefix:                  "prefix",
+		HostnameSuffix:             "mesh",
+		SyncNamespace:              "ns",
+		CacheRefreshDuration:       time.Minute,
+		ClusterRegistriesNamespace: "default",
+		DependenciesNamespace:      "default",
+		SecretResolver:             "",
+	}
+
+	p.LabelSet.WorkloadIdentityKey = "identity"
+	p.LabelSet.EnvKey = "admiral.io/env"
+	p.LabelSet.GlobalTrafficDeploymentLabel = "identity"
+
+	registry, _ := InitAdmiral(context.Background(), p)
+
+	handler := RoutingPolicyHandler{}
+
+	rpFilterCache := &routingPolicyFilterCache{}
+	rpFilterCache.filterCache = make(map[string]map[string]map[string]string)
+	rpFilterCache.mutex = &sync.Mutex{}
+
+
+	routingPolicyController := &admiral.RoutingPolicyController{IstioClient: istiofake.NewSimpleClientset()}
+	remoteController, _ := createMockRemoteController(func(i interface{}) {
+
+	})
+	remoteController.RoutingPolicyController = routingPolicyController
+
+	registry.RemoteControllers = map[string]*RemoteController{"cluster-1": remoteController}
+	registry.AdmiralCache.RoutingPolicyFilterCache = rpFilterCache
+
+	// foo is dependent upon bar and bar has a deployment in the same cluster.
+	registry.AdmiralCache.IdentityDependencyCache.Put("foo", "bar", "bar")
+	registry.AdmiralCache.IdentityClusterCache.Put("bar", remoteController.ClusterID, remoteController.ClusterID)
+
+
+	// foo is also dependent upon bar2 but bar2 is in a different cluster, so this cluster should not have the envoyfilter created
+	registry.AdmiralCache.IdentityDependencyCache.Put("foo", "bar2", "bar2")
+	registry.AdmiralCache.IdentityClusterCache.Put("bar2", "differentCluster", "differentCluster")
+
+	// foo1 is dependent upon bar 1 but bar1 does not have a deployment so it is missing from identityClusterCache
+	registry.AdmiralCache.IdentityDependencyCache.Put("foo1", "bar1", "bar1")
+
+
+	handler.RemoteRegistry = registry
+
+	routingPolicyFoo := &v1.RoutingPolicy{
+		TypeMeta:   time2.TypeMeta{},
+		ObjectMeta: time2.ObjectMeta{
+			Labels: map[string]string{
+				"identity": "foo",
+				"admiral.io/env": "stage",
+			},
+		},
+		Spec:       model.RoutingPolicy{
+			Plugin:               "test",
+			Hosts:                []string{"e2e.testservice.mesh"},
+			Config: map[string]string{
+				"cachePrefix": "cache-v1",
+				"cachettlSec": "86400",
+				"routingServiceUrl": "e2e.test.routing.service.mesh",
+				"pathPrefix": "/sayhello,/v1/company/{id}/",
+			},
+		},
+		Status:     v1.RoutingPolicyStatus{},
+	}
+
+	routingPolicyFoo1 := routingPolicyFoo.DeepCopy()
+	routingPolicyFoo1.Labels[common.GetWorkloadIdentifier()] = "foo1"
+
+
+	testCases := []struct {
+		name 						string
+		routingPolicy				*v1.RoutingPolicy
+		expectedFilterCacheKey		string
+		valueExpected				bool
+
+	} {
+		{
+			name:                     "should have the right envoyFilter in filter cache",
```

Comment:
To enhance readability, will be useful to add the current state, and what is expected. (like mentioned in others - `If dependent deployment does not exist, the filter should not be created`)